### PR TITLE
lief, python3Packages.lief: 0.12.3 -> 0.13.2

### LINKED ISSUE #249419: Build failure: lief
### Steps To Reproduce

1. `nix build nixpkgs#lief`

### Build log

https://gist.github.com/GenericNerdyUsername/6e19169a078661b7bf629d4b997fd236

### Additional context

Might be worth updating to 0.13.2 as well

### Notify maintainers

@Lassulus 

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.4.7, NixOS, 23.11 (Tapir), 23.11.20230805.0d2fb29`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.15.1`
 - channels(root): `"nixos"`
 - channels(mia): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### PR DESCRIPTION
**Pull request by GenericNerdyUsername**
## Description of changes

Fix build and update for lief.

Closes #249419.

@Lassulus 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/lief/default.nix [hunk 1]
```diff
@@ -31,40 +33,23 @@ stdenv.mkDerivation rec {
...
+    popd
   '';
 
   meta = with lib; {
     description = "Library to Instrument Executable Formats";
     homepage = "https://lief.quarkslab.com/";
     license = [ licenses.asl20 ];
     platforms = with platforms; linux ++ darwin;
-    maintainers = [ maintainers.lassulus ];
+    maintainers = with maintainers; [ genericnerdyusername ];
```
Comment (matthiasbeyer) [inline pkgs/development/libraries/lief/default.nix:53]: I think this should be a separate commit!

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]

**Review by matthiasbeyer — APPROVED** [review]

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 251414` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>lief</li>
    <li>lief.py (python310Packages.lief ,python310Packages.lief.py)</li>
    <li>python311Packages.lief (python311Packages.lief.py)</li>
  </ul>
</details>

**Comment by fzakaria** [comment]
@GenericNerdyUsername this doesn't set the GIT_COMMIT_HASH in the version you get something like:
```
>>> import lief;print(lief.__version__)
0.13.2-
```

I'm not sure how to set this variable: https://github.com/lief-project/LIEF/blob/3414ded8cdcbd9705f7871c66c212b15cd74ea69/cmake/LIEFGit.cmake#L10

But that's where it is set.

**Comment by fzakaria** [comment]
I had to do this in the meantime: https://github.com/fzakaria/sqlelf/pull/8
